addons/account/wizard/pos_box.py

```diff
@@ -33,8 +33,8 @@ def _create_bank_statement_line(self, record):
             if record.state == 'confirm':
                 raise UserError(_("You cannot put/take money in/out for a bank statement which is closed."))
             values = box._calculate_values_for_statement_line(record)
-            account = record.journal_id.company_id.transfer_account_id
-            self.env['account.bank.statement.line'].with_context(counterpart_account_id=account.id).create(values)
+            account = record.journal_id.payment_debit_account_id
```

Comment:
You should use a different account if either it is a cash in (Oustanding Receipts) our cash out (Outstanding Payment)
```suggestion
            account = record.journal_id.payment_debit_account_id if values['amount'] > 0 else record.journal_id.payment_credit_account_id
```